src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -2554,15 +2555,14 @@ private RefKind GetEffectiveParameterRefKind(ParameterSymbol parameter, RefKind
             var refs = ArrayBuilder<RefKind>.GetInstance();
             bool anyRef = false;
             var parameters = member.GetParameters();
-            var elementType = ((ArrayTypeSymbol)parameters[parameters.Length - 1].Type).ElementType;
             bool hasAnyRefArg = argumentRefKinds.Any();
             hasAnyRefOmittedArgument = false;
 
             for (int arg = 0; arg < argumentCount; ++arg)
             {
                 var parm = argToParamMap.IsDefault ? arg : argToParamMap[arg];
                 var parameter = parameters[parm];
-                types.Add(parm == parameters.Length - 1 ? elementType : parameter.Type);
+                types.Add(parameter.IsParams ? ((ArrayTypeSymbol)parameter.Type).ElementType : parameter.Type);
```

Comment:
>types.Add(parameter.IsParams ? ((ArrayTypeSymbol)parameter.Type).ElementType : parameter.Type); [](start = 16, length = 95)

This doesn't feel right. I see several problems with this code:
- only the last parameters should be considered for expansion.
- if it is not the last parameter, we probably cannot assume its type is an array type and the array is of the right kind.